src/Bicep.Types/Concrete/ResourceType.cs

```diff
@@ -11,15 +11,51 @@ namespace Azure.Bicep.Types.Concrete
     public enum ResourceFlags
     {
         None = 0,
-        ReadOnly = 1 << 0,
-        WriteOnly = 1 << 1,
+        ReadOnly = 1 << 0
     }
 
     public class ResourceType : TypeBase
     {
         [JsonConstructor]
-        public ResourceType(string name, ScopeType scopeType, ScopeType? readOnlyScopes, ITypeReference body, ResourceFlags flags, IReadOnlyDictionary<string, ResourceTypeFunction>? functions)
-            => (Name, ScopeType, ReadOnlyScopes, Body, Flags, Functions) = (name, scopeType, readOnlyScopes, body, flags, functions);
+        public ResourceType(
+            string name, 
+            ScopeType scopeType, 
```

Comment:
I would mark this as nullable. I think a `JsonConstructor` will treat any non-nullable argument as a required property